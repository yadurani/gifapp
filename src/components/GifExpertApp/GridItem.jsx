import React from 'react';
import PropTypes from 'prop-types';

const GridItem = ({ url, title }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title}/>
    </div>
  )
};

GridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}

export default GridItem
