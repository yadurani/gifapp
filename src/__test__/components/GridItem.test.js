import React from 'react'
import { shallow } from 'enzyme';

import GridItem from '../../components/GifExpertApp/GridItem';

describe('Probando el componente <GridItem />', () => {
  const title = 'Un titulo'
  const url = 'https://localhost/image.png'
  const wrapper = shallow(<GridItem title={title} url={url}/>)

  it('Debe hacer toMatchSnapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Debe de tener una imagen con una url', () => {
    const img = wrapper.find('img')

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  })

  it('Debe tener animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');

    expect(className.includes('animate__fadeIn')).toBe(true)
  })
})