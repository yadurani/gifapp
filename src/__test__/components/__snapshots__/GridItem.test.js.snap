// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GridItem /> Debe hacer toMatchSnapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/image.png"
  />
</div>
`;
